pkg/dataServer/dataServerOps.go

```diff
@@ -98,3 +67,9 @@ func (ops *DataServerOps) Start(ctx context.Context) error {
 func (ops *DataServerOps) Ready() chan bool {
 	return ops.server.Ready()
 }
+
+// Stop will take care of stopping the dataserver component.
+func (ops *DataServerOps) Stop() {
+	ops.dataServerExitCh <- true
```

Comment:
lets use the context to indicate that need to exit.
```
ctx,exit = context.WithCancel(ctx)
```
and when you want to exit you just call:
```
exit()
```

Reply:
Done